sql/tests: TestRandomSyntaxSQLSmith failed [Parse followed by Format is not idempotent]

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20561509?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20561509?buildTab=artifacts#/) on master @ [ae0c26a5eb2333c0ae3d235b43d4e680db0ae453](https://github.com/cockroachdb/cockroach/commits/ae0c26a5eb2333c0ae3d235b43d4e680db0ae453):


```
    rsg_test.go:786: CREATE TABLE table😻_1 (col1_0 OID NULL, col1_1 NAME NULL, c̺ol1_2 STRING COLLATE en_US, col1_3 STRING COLLATE en_US NOT NULL, c̦ol1_4 INT4 NOT NULL, col1_5 STRING COLLATE de_DE NOT NULL, col1_6 REGROLE, col1_7 INT2 NOT NULL, "col\\U0010027C1_8" STRING NULL, col1_9 TIMETZ NOT NULL, col1_10 BYTES NOT NULL, "%18co
        l1_11" UUID NOT NULL, col1_12 BOOL NOT NULL, col1_13 VECTOR NOT NULL, col1_14 TSVECTOR NOT NULL, "col'1 	_15" STRING NULL AS (lower(col1_1)) STORED, ")c%qol1_16" STRING AS (lower(CAST(c̺ol1_2 AS STRING))) STORED, "col&1_17" INT8 AS (c̦ol1_4 + col1_7) VIRTUAL, PRIMARY KEY (col1_7 DESC), INDEX (c̦ol1_4, col1_1 ASC), INVERTED INDEX (")c%qol1_16" gin_trgm_ops ASC) WHERE (((((((table😻_1.")c%qol1_16" < e'\U00002603':::STRING) OR (table😻_1."col'1 	_15" > '"':::STRING)) OR (table😻_1.c̦ol1_4 < 1:::INT8)) OR (NOT table😻_1.col1_12)) AND (table😻_1.col1_9 != '00:00:00+15:59':::TIMETZ)) OR (table😻_1.col1_10 = '\xff':::BYTES)) AND (table😻_1."col\\U0010027C1_8" < 'X':::STRING)) OR (table😻_1."col&1_17" > (-32768):::INT8) NOT VISIBLE, FAMILY (col1_14), FAMILY (col1_6), FAMILY (c̺ol1_2, ")c%qol1_16", col1_1), FAMILY (col1_3, col1_0), FAMILY ("col\\U0010027C1_8"), FAMILY (col1_10, col1_9), FAMILY (col1_13), FAMILY (col1_5, "%18co
        l1_11", col1_7), FAMILY (col1_12), FAMILY (c̦ol1_4), FAMILY ("col'1 	_15"));
    rsg_test.go:786: ALTER TABLE table😻_1 INJECT STATISTICS e'[{"avg_size": 2, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 971257476896886153, "histo_col_type": "", "name": "__auto__", "null_count": 483371070342044391, "row_count": 2182208356574530318}, {"avg_size": 20, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1679064647078782440, "histo_col_type": "", "name": "__auto__", "null_count": 1657402104029962202, "row_count": 2182208356574530318}, {"avg_size": 0, "columns": ["c\u033Aol1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1521978945901524218, "histo_col_type": "", "name": "__auto__", "null_count": 1568758643545043944, "row_count": 2182208356574530318}, {"avg_size": 19, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 402587131045599018, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 24, "columns": ["c\u0326ol1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1373285590794430251, "histo_buckets": [{"distinct_range": 0, "num_eq": 7531301964944746664, "num_range": 0, "upper_bound": "-1"}, {"distinct_range": 0, "num_eq": 900000000, "num_range": 5000, "upper_bound": "-10129157"}, {"distinct_range": 2610096079.0551577, "num_eq": 900000000, "num_range": 7000000000, "upper_bound": "-76483083"}, {"distinct_range": 0, "num_eq": 4284786258172844485, "num_range": 7000, "upper_bound": "-1233949206"}, {"distinct_range": 1204070228.523799, "num_eq": 10000000, "num_range": 5000000000, "upper_bound": "-704071134"}, {"distinct_range": 6049807709509935000, "num_eq": 5000000, "num_range": 6187123031332123886, "upper_bound": "1105130729"}], "histo_col_type": "INT4", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 8, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 753127714156182876, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 21, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 461838137772792148, "histo_col_type": "", "name": "__auto__", "null_count": 985929492414875203, "row_count": 2182208356574530318}, {"avg_size": 5, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1016520795351509268, "histo_col_type": "INT2", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 29, "columns": ["\\"col\\\\\\\\U0010027C1_8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 615532454360712029, "histo_col_type": "", "name": "__auto__", "null_count": 1754230521655163652, "row_count": 2182208356574530318}, {"avg_size": 27, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 521377319085355953, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 10, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 786176442374949298, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 4, "columns": ["\\"%18co\\nl1_11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1314747159516769207, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 11, "columns": ["col1_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 117265374277790229, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 23, "columns": ["col1_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1784388760684448486, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 18, "columns": ["col1_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1448366556176446560, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2182208356574530318}, {"avg_size": 22, "columns": ["\\"col\'1 \\t_15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1618600551578625048, "histo_col_type": "", "name": "__auto__", "null_count": 1638136794305741061, "row_count": 2182208356574530318}, {"avg_size": 12, "columns": ["\\")c%qol1_16\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1600493091910407341, "histo_buckets": [{"distinct_range": 0, "num_eq": 8026165718404300533, "num_range": 0, "upper_bound": "\'"}, {"distinct_range": 69236.30942003972, "num_eq": 20000, "num_range": 70000, "upper_bound": "Q4q"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 1714823330860356674, "row_count": 2182208356574530318}, {"avg_size": 15, "columns": ["\\"col&1_17\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1713803738397020836, "histo_col_type": "", "name": "__auto__", "null_count": 455499534239649818, "row_count": 2182208356574530318}]':::JSONB;
    rsg_test.go:786: CREATE TYPE rand_typ_0 AS (ezrjg BOX2D, dz BYTES, x BIT(38), wgim STRING, ecd FLOAT8);
    rsg_test.go:786: CREATE TYPE rand_typ_1 AS ENUM ('ncov', 'pg');
    rsg_test.go:786: CREATE TYPE rand_typ_2 AS (cgpru TIMESTAMPTZ);
    rsg_test.go:786: SET sql_safe_updates = false;;
    rsg_test.go:786: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (693270248611500889);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (313891944675391506);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (4457434568239615047);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (2828224010621068516);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (3870728648926050383);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (9129669684245644146);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (580521415881534171);
    rsg_test.go:786: ALTER TABLE seed SCATTER;
    rsg_test.go:788: 
    rsg_test.go:789: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith631520955
--- FAIL: TestRandomSyntaxSQLSmith (305.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154601 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-foundations branch-release-24.1 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55206

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20562560?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20562560?buildTab=artifacts#/) on master @ [f520554bbd8f278a97a0bef459f7f671c01f4564](https://github.com/cockroachdb/cockroach/commits/f520554bbd8f278a97a0bef459f7f671c01f4564):


```
    rsg_test.go:786: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:786: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:786: SET CLUSTER SETTING sql.stats.forecasts.enabled = false;;
    rsg_test.go:786: CREATE TABLE "'table_1" ("co l1_0" TIMESTAMP NULL, "col%q1_1" STRING COLLATE da_DK, col1_2 "char" NOT NULL, col1_3 INT8, col1_4 INT8 NOT NULL, "col1 _5" GEOGRAPHY NULL, "col\\xb51_6" TIMESTAMPTZ NOT NULL, col1_7 TIMESTAMPTZ NULL, col1_8 REGROLE, col1_9 GEOGRAPHY NOT NULL, col1_10 INT8 NOT NULL, col1_11 BPCHAR NULL, "cOl1_12" VARCHAR NOT NULL, "c \\U0001932Eol1_13" INT8 AS ((col1_3 + col1_4) + col1_10) VIRTUAL, col1_14 STRING AS (lower(CAST("col%q1_1" AS STRING))) VIRTUAL, "%qcol1_15" INT8 AS ((col1_3 + col1_10) + col1_4) VIRTUAL, col1_16 INT8 NOT NULL AS (col1_3 + col1_4) STORED, PRIMARY KEY (col1_16 ASC), UNIQUE (col1_7) STORING ("co l1_0", col1_3, "col1 _5", col1_8, "cOl1_12") WHERE (((((((((((("'table_1"."co l1_0" < '294276-12-31 23:59:59.999999':::TIMESTAMP) AND ("'table_1"."col\\xb51_6" <= '0001-01-01 00:00:00+00':::TIMESTAMPTZ)) OR ("'table_1"."%qcol1_15" = (-128):::INT8)) AND ("'table_1".col1_16 < 1:::INT8)) OR ("'table_1".col1_3 > (-32768):::INT8)) OR ("'table_1".col1_2 >= e'\x00':::STRING)) OR ("'table_1".col1_7 != '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ)) AND ("'table_1".col1_10 >= (-9223372036854775807):::INT8)) AND ("'table_1"."cOl1_12" >= e'\U00002603':::STRING)) OR ("'table_1".col1_4 <= 2147483647:::INT8)) AND ("'table_1".col1_14 >= e'\U00002603':::STRING)) OR ("'table_1"."c \\U0001932Eol1_13" != 32767:::INT8)) OR ("'table_1".col1_11 = '':::STRING), INDEX (col1_2) STORING (col1_3, col1_8, col1_9, col1_11, "cOl1_12") WHERE (((("'table_1".col1_2 != '':::STRING) OR ("'table_1".col1_10 < 127:::INT8)) AND ("'table_1".col1_11 < 'X':::STRING)) AND ("'table_1"."col\\xb51_6" <= '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ)) AND ("'table_1"."%qcol1_15" != (-128):::INT8), INDEX (col1_7 DESC, col1_4) PARTITION BY LIST (col1_7) (PARTITION "'table_1_part)_0" VALUES IN (('1970-10-21 16:46:19.000061+00':::TIMESTAMPTZ,)), PARTITION "'table)_1_part_1" VALUES IN (('1971-11-25 15:38:36.000908+00':::TIMESTAMPTZ,)), PARTITION "'table_1_p%art_2" VALUES IN (('3000-01-01 00:00:00+00':::TIMESTAMPTZ,)), PARTITION "'table_1_part_3" VALUES IN (('2000-01-25 16:39:03.000972+00':::TIMESTAMPTZ,)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))), INDEX (col1_4) STORING ("co l1_0", col1_2, "col1 _5", col1_9, "cOl1_12"), INDEX (col1_4 ASC) STORING (col1_3, col1_8, col1_9, col1_10, col1_11, "cOl1_12") PARTITION BY LIST (col1_4) (PARTITION "'table_1_part_0" VALUES IN ((829143120805402441:::INT8,), (2760372109848538823:::INT8,), (2544275663464594803:::INT8,), (1879664724426742410:::INT8,), ((-4610613687770730188):::INT8,), (2612785186447449473:::INT8,), ((-7403120790630317718):::INT8,), (527880717650804212:::INT8,)), PARTITION "'table_1_part_1" VALUES IN ((1145336162060468978:::INT8,), (4964385868533855338:::INT8,), ((-8335628668919990393):::INT8,), ((-5940528321799281503):::INT8,), (7601101074245487444:::INT8,), (4002023308883334212:::INT8,), ((-2178226745693340285):::INT8,), (8120004631527528935:::INT8,)), PARTITION "'table_1 _part_2" VALUES IN (((-6654181494381638347):::INT8,), ((-6950245683311823496):::INT8,), (8097636690272012803:::INT8,), ((-4449675270878475522):::INT8,), ((-6773541311390754377):::INT8,), ((-684731467700676778):::INT8,), (3435838498581437464:::INT8,), (8460172779132844503:::INT8,)), PARTITION "'ta̩ble_1_part_3" VALUES IN ((2092178245036332409:::INT8,), (5883507770829363444:::INT8,), ((-2927504655969222589):::INT8,), (8832070021653976827:::INT8,), (2408378643140540508:::INT8,), ((-3553659161965279266):::INT8,), ((-3560353318803871162):::INT8,), (0:::INT8,)), PARTITION "'table_1_part_4" VALUES IN ((5533859559961426204:::INT8,), ((-5290363499554529754):::INT8,), (3324992150732993870:::INT8,), (7307284108735880128:::INT8,), ((-7098111961024756554):::INT8,), ((-8363028127968622685):::INT8,), ((-7726777868662677559):::INT8,), (6620448852579734126:::INT8,)), PARTITION "'table_1_part_5" VALUES IN ((1374905029693893253:::INT8,), (5157175707890254298:::INT8,), (6795313906352100704:::INT8,), ((-7377299086102350917):::INT8,), (3316501741365699676:::INT8,), ((-7179567428787174486):::INT8,), (6296047617430244522:::INT8,), ((-7236006837899370422):::INT8,)), PARTITION "'table_1_part _6" VALUES IN ((9223372036854775807:::INT8,), (1212972854650214368:::INT8,), (187543615707623587:::INT8,), (427993888779733968:::INT8,), ((-6182810825362476979):::INT8,), ((-789930921967500545):::INT8,), ((-4969047503158718411):::INT8,), (5466796120971073145:::INT8,)), PARTITION "'table_""1_part_7" VALUES IN (((-2573974550016772922):::INT8,), (1163488480844592495:::INT8,), (5358741733890097833:::INT8,), ((-5862848371857487779):::INT8,), (5148777863457248179:::INT8,), ((-7079712756774441429):::INT8,), ((-4202121128312916011):::INT8,), ((-2313702200770241029):::INT8,)), PARTITION "'table_1_part_8" VALUES IN (((-1055456339387741094):::INT8,), (2748618269138017282:::INT8,), (8063431806882203416:::INT8,), (399006990257708417:::INT8,), ((-6544790603781928962):::INT8,), (5925546818396732573:::INT8,), (8550536839033205044:::INT8,), ((-4826163654981854089):::INT8,)), PARTITION "'table\\u561F_1_part_9" VALUES IN ((8888549933856704972:::INT8,), ((-7325258723664957715):::INT8,), (1832082659821678765:::INT8,), ((-5992270350548870555):::INT8,), ((-128):::INT8,), ((-5530573873049745256):::INT8,), ((-7489424800285473393):::INT8,), ((-4136152837759982069):::INT8,)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))) NOT VISIBLE);
    rsg_test.go:786: CREATE TABLE "\\x1f😹table_2" (col2_0 INET NULL, "%vcOl2�_1" NAME NOT NULL, col2_2 REGTYPE NULL, "c\\x42 ol2_3" GEOMETRY NOT NULL, col2_4 STRING NULL AS (lower(CAST(col2_0 AS STRING))) VIRTUAL, INDEX (col2_4 DESC) STORING (col2_0, "%vcOl2�_1") NOT VISIBLE, UNIQUE (col2_2 DESC) STORING (col2_0, "%vcOl2�_1", "c\\x42 ol2_3") WHERE ("\\x1f😹table_2".col2_4 < '':::STRING) OR ("\\x1f😹table_2"."%vcOl2�_1" < '':::STRING:::NAME), INDEX (col2_0 DESC) WHERE ("\\x1f😹table_2"."%vcOl2�_1" <= e'\'':::STRING:::NAME) OR ("\\x1f😹table_2".col2_4 = e'\U00002603':::STRING) NOT VISIBLE, INDEX (col2_0) WHERE ("\\x1f😹table_2".col2_4 <= '':::STRING) OR ("\\x1f😹table_2"."%vcOl2�_1" < e'\x00':::STRING:::NAME) NOT VISIBLE, INDEX (lower(CAST(col2_0 AS STRING)) DESC) STORING (col2_0, "%vcOl2�_1", col2_2, "c\\x42 ol2_3"));
    rsg_test.go:786: ALTER TABLE "'table_1" INJECT STATISTICS e'[{"avg_size": 26, "columns": ["\\"co l1_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 21488412280434040, "histo_col_type": "", "name": "__auto__", "null_count": 63457964747106278, "row_count": 81315883018247050}, {"avg_size": 4, "columns": ["\\"col%q1_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 73688725146817743, "histo_col_type": "", "name": "__auto__", "null_count": 26028117610713989, "row_count": 81315883018247050}, {"avg_size": 28, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 12200932566955197, "histo_col_type": "\\"char\\"", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}, {"avg_size": 3, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 23982512671243304, "histo_col_type": "", "name": "__auto__", "null_count": 48450110918519744, "row_count": 81315883018247050}, {"avg_size": 27, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 11838480504283321, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}, {"avg_size": 19, "columns": ["\\"col1 _5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 74937855293756059, "histo_col_type": "", "name": "__auto__", "null_count": 30004850710739247, "row_count": 81315883018247050}, {"avg_size": 26, "columns": ["\\"col\\\\\\\\xb51_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 67563751535213292, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}, {"avg_size": 26, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8683536427992913, "histo_buckets": [{"distinct_range": 0, "num_eq": 4379476891844060045, "num_range": 0, "upper_bound": "1977-11-08 21:56:38.00088+00"}, {"distinct_range": 71953856531215690, "num_eq": 4009181204728288383, "num_range": 85814775008340306, "upper_bound": "1998-02-28 00:13:48.00033+00"}], "histo_col_type": "TIMESTAMPTZ", "histo_version": 3, "name": "__auto__", "null_count": 43798597394025653, "row_count": 81315883018247050}, {"avg_size": 12, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 35042233838959052, "histo_col_type": "", "name": "__auto__", "null_count": 15891494670551720, "row_count": 81315883018247050}, {"avg_size": 15, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 56062729955723620, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}, {"avg_size": 24, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 23250872135116326, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}, {"avg_size": 12, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 25058314742032685, "histo_col_type": "", "name": "__auto__", "null_count": 29304740127757560, "row_count": 81315883018247050}, {"avg_size": 29, "columns": ["\\"cOl1_12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 71723656259374613, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}, {"avg_size": 28, "columns": ["\\"c \\\\\\\\U0001932Eol1_13\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 74785524306534160, "histo_col_type": "", "name": "__auto__", "null_count": 34833197564706005, "row_count": 81315883018247050}, {"avg_size": 7, "columns": ["col1_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 36530517189548539, "histo_col_type": "", "name": "__auto__", "null_count": 25047150750110212, "row_count": 81315883018247050}, {"avg_size": 30, "columns": ["\\"%qcol1_15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1104239965773417, "histo_col_type": "", "name": "__auto__", "null_count": 71321220949154973, "row_count": 81315883018247050}, {"avg_size": 14, "columns": ["col1_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 75081576412728781, "histo_col_type": "INT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 81315883018247050}]':::JSONB;
    rsg_test.go:786: ALTER TABLE "\\x1f😹table_2" INJECT STATISTICS e'[{"avg_size": 12, "columns": ["col2_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 606558372182051552, "histo_buckets": [{"distinct_range": 0, "num_eq": 613682106586240808, "num_range": 0, "upper_bound": "253.71.212.242/1"}, {"distinct_range": 0, "num_eq": 10000000000, "num_range": 607683539467377455, "upper_bound": "33.217.186.155/10"}, {"distinct_range": 0, "num_eq": 8261085698121696492, "num_range": 5730008064730476854, "upper_bound": "86.180.102.48/19"}, {"distinct_range": 3358259330929646000, "num_eq": 3000, "num_range": 5943339610915100292, "upper_bound": "fc51:49ac:7398:8ec5:4b7d:ef47:8cbc:b355/115"}, {"distinct_range": 781240584848423300, "num_eq": 7323045614118366121, "num_range": 1869688379579222924, "upper_bound": "ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff"}], "histo_col_type": "INET", "histo_version": 3, "name": "__auto__", "null_count": 688158561091500128, "row_count": 3699004472667192737}, {"avg_size": 1, "columns": ["\\"%vcOl2\\u000c_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 812219317342750344, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3699004472667192737}, {"avg_size": 27, "columns": ["col2_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3076544498151801161, "histo_buckets": [{"distinct_range": 0, "num_eq": 6402291609926441951, "num_range": 0, "upper_bound": "445250309"}, {"distinct_range": 0, "num_eq": 3000, "num_range": 60, "upper_bound": "3580979144"}, {"distinct_range": 0, "num_eq": 7637016998033549178, "num_range": 10000000, "upper_bound": "1038649630"}, {"distinct_range": 316715555535596800, "num_eq": 316588927024654444, "num_range": 8260048633079562425, "upper_bound": "1725305257"}, {"distinct_range": 0, "num_eq": 435956574920741512, "num_range": 3000000, "upper_bound": "2849939181"}], "histo_col_type": "REGTYPE", "histo_version": 3, "name": "__auto__", "null_count": 2187701626258562702, "row_count": 3699004472667192737}, {"avg_size": 11, "columns": ["\\"c\\\\\\\\x42 ol2_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2066211736353263565, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3699004472667192737}, {"avg_size": 14, "columns": ["col2_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1419233694413540548, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": ""}, {"distinct_range": 10000, "num_eq": 6281398993668991352, "num_range": 10000, "upper_bound": "^u"}, {"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "j8{"}, {"distinct_range": 0, "num_eq": 100000000000, "num_range": 4000000, "upper_bound": "iZ$Y\\u0012\\r9"}, {"distinct_range": 0, "num_eq": 1000000, "num_range": 5407444044706653407, "upper_bound": "ejnLuh\\r\\u0016"}, {"distinct_range": 0, "num_eq": 8000000, "num_range": 9163607993958170452, "upper_bound": "j,\\u0012Yfmi\\u0011"}, {"distinct_range": 5718442441514350000, "num_eq": 5385694941912673161, "num_range": 5718442441514349562, "upper_bound": "\\u0011\\u000fm~?D\\u000f{b"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 2458174665685397720, "row_count": 3699004472667192737}]':::JSONB;
    rsg_test.go:786: CREATE TYPE rand_typ_0 AS ENUM ('n', 'e', 'wfgfz', 'y', 'monrup');
    rsg_test.go:786: CREATE TYPE rand_typ_1 AS ENUM ('am', 'zh', 'b');
    rsg_test.go:786: SET sql_safe_updates = false;;
    rsg_test.go:786: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (1888705314775223427);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (2261389166410066495);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (5113556520803793618);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (521793218214824337);
    rsg_test.go:786: ALTER TABLE seed SPLIT AT VALUES (1604321806674283908);
    rsg_test.go:786: ALTER TABLE seed SCATTER;
    rsg_test.go:788: 
    rsg_test.go:789: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith4037227993
--- FAIL: TestRandomSyntaxSQLSmith (302.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154601 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-foundations branch-release-24.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20569436?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20569436?buildTab=artifacts#/) on master @ [aa46307cc6478d031cbff4c65be2df5950fae2d8](https://github.com/cockroachdb/cockroach/commits/aa46307cc6478d031cbff4c65be2df5950fae2d8):


```
        	20:::INT8
    rsg_test.go:114: Last executed (49): SELECT NULL AS "co%d9l_295239", NULL AS col_295240 LIMIT 5:::INT8
    rsg_test.go:117: Currently active statements: map[]
    rsg_test.go:784: To reproduce, use schema:
    rsg_test.go:786: SET sql_safe_updates = false;;
    rsg_test.go:786: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:786: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:786: SET CLUSTER SETTING sql.stats.forecasts.enabled = false;;
    rsg_test.go:786: CREATE TABLE table_1 ("%qcol1_0" TIME NOT NULL, "col 1_1" BYTES NOT NULL, col1_2 BIT(34) NOT NULL, "*col 1_3" TSVECTOR NOT NULL, col1_4 INT4 NOT NULL, "
col1_5" NAME NOT NULL, "coL1_6" NAME NOT NULL, col1_7 BOOL, "\\x0dcol1_8" OID, 😖col1_9 "char" NOT NULL, "col 1_10" INT4 NOT NULL AS (abs(col1_4)) STORED, col1_11 STRING AS (lower(CAST("\\x0dcol1_8" AS STRING))) STORED, "col1|_12" INT4 NOT NULL AS (abs(col1_4)) VIRTUAL, PRIMARY KEY (😖col1_9, "%qcol1_0" ASC, "col 1_1" DESC));
    rsg_test.go:786: CREATE TABLE table_2 (co😖l2_0 FLOAT8 NULL, col2_1 JSONB, "co%l2_2" NAME NOT NULL, "coL'2_3" UUID NOT NULL, col2_4 JSONB NOT NULL, "c%pol2
_5" BPCHAR NOT NULL, "co%pl)2_6" PG_LSN NULL, col2_7 STRING AS (lower(CAST(col2_1 AS STRING))) VIRTUAL, PRIMARY KEY (col2_4, "co%l2_2" DESC, "coL'2_3"), INDEX (col2_7 DESC, "co%l2_2" DESC, col2_4 DESC), UNIQUE ("c%pol2
_5" ASC, lower(CAST(col2_1 AS STRING)) DESC, col2_7 ASC, col2_4) STORING (co😖l2_0) WHERE (table_2."co%l2_2" <= e'\U00002603':::STRING:::NAME) AND (table_2.col2_7 != '':::STRING), FAMILY ("c%pol2
_5", col2_4), FAMILY ("coL'2_3"), FAMILY ("co%l2_2"), FAMILY (co😖l2_0), FAMILY ("co%pl)2_6"), FAMILY (col2_1));
    rsg_test.go:786: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 23, "columns": ["\\"%qcol1_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2707311449385158506, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 30, "columns": ["\\"col 1_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2588097873785955725, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 2, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2190733749792177834, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 10, "columns": ["\\"*col 1_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 846863764582866717, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 30, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3657295158363316605, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 8, "columns": ["\\"\\rcol1_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1151484692214566221, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 28, "columns": ["\\"coL1_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 841202005419683691, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 9, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3094930991115356752, "histo_col_type": "", "name": "__auto__", "null_count": 133479321369632812, "row_count": 4313403586651645955}, {"avg_size": 12, "columns": ["\\"\\\\\\\\x0dcol1_8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1752510616740498387, "histo_col_type": "", "name": "__auto__", "null_count": 199773513981757481, "row_count": 4313403586651645955}, {"avg_size": 30, "columns": ["\U0001F616col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 425360207609827767, "histo_buckets": [{"distinct_range": 0, "num_eq": 8000000, "num_range": 0, "upper_bound": "E"}, {"distinct_range": 1992871218591071000, "num_eq": 0, "num_range": 1992871218591070950, "upper_bound": "X"}, {"distinct_range": 295062833427861000, "num_eq": 7938606305919078661, "num_range": 790996377448706853, "upper_bound": "g"}], "histo_col_type": "\\"char\\"", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 25, "columns": ["\\"col 1_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1338917011090325138, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}, {"avg_size": 9, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3599153943016201667, "histo_col_type": "", "name": "__auto__", "null_count": 4296620936643287190, "row_count": 4313403586651645955}, {"avg_size": 14, "columns": ["\\"col1|_12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3392507212585023241, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4313403586651645955}]':::JSONB;
    rsg_test.go:786: ALTER TABLE table_2 INJECT STATISTICS e'[{"avg_size": 27, "columns": ["co\U0001F616l2_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1977206968382406866, "histo_col_type": "", "name": "__auto__", "null_count": 6776816108427191416, "row_count": 7243032468914788925}, {"avg_size": 16, "columns": ["col2_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1786917687027342278, "histo_col_type": "", "name": "__auto__", "null_count": 6078052963938378002, "row_count": 7243032468914788925}, {"avg_size": 18, "columns": ["\\"co%l2_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4603416378031128163, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 7243032468914788925}, {"avg_size": 31, "columns": ["\\"coL\'2_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4261419010888581249, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 7243032468914788925}, {"avg_size": 12, "columns": ["col2_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5412037244790647384, "histo_buckets": [{"distinct_range": 0, "num_eq": 3279252074366472530, "num_range": 0, "upper_bound": "[[{\\"bar\\": false}, false], [\\"RJxk\\"], 2.150354971296231, false]"}], "histo_col_type": "JSONB", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7243032468914788925}, {"avg_size": 17, "columns": ["\\"c%pol2\\r_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7078908055063724208, "histo_buckets": [{"distinct_range": 0, "num_eq": 1000000, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 6941052995109760694, "num_range": 3512037799125615032, "upper_bound": "!"}, {"distinct_range": 5440276867937642000, "num_eq": 4000000, "num_range": 6544463760424595109, "upper_bound": "*"}, {"distinct_range": 0, "num_eq": 20000, "num_range": 0, "upper_bound": ":"}, {"distinct_range": 0, "num_eq": 30000, "num_range": 100000, "upper_bound": "="}, {"distinct_range": 0, "num_eq": 5737785564781725528, "num_range": 3541715143499317309, "upper_bound": "D"}, {"distinct_range": 40000000, "num_eq": 0, "num_range": 40000000, "upper_bound": "X"}, {"distinct_range": 7.6904329409972325, "num_eq": 3000, "num_range": 10, "upper_bound": "\U00002603"}], "histo_col_type": "BPCHAR", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7243032468914788925}, {"avg_size": 14, "columns": ["\\"co%pl)2_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5961326340072493474, "histo_col_type": "", "name": "__auto__", "null_count": 1060123670588658321, "row_count": 7243032468914788925}, {"avg_size": 5, "columns": ["col2_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6147168038728605174, "histo_buckets": [{"distinct_range": 0, "num_eq": 1392637867783725741, "num_range": 0, "upper_bound": ""}, {"distinct_range": 7615278111362775000, "num_eq": 80000000000, "num_range": 7615278111362775405, "upper_bound": "\\u0000"}, {"distinct_range": 4634333264773830000, "num_eq": 1000, "num_range": 4634333264773829304, "upper_bound": "\\u001c\\u000c"}, {"distinct_range": 3192783123901542400, "num_eq": 2714819661910113695, "num_range": 3192783123901542270, "upper_bound": "\\u0017%b\\u0016"}, {"distinct_range": 0, "num_eq": 4295380368932215739, "num_range": 400000, "upper_bound": "$\\u001e.\\u000cM"}, {"distinct_range": 0, "num_eq": 391291041546995055, "num_range": 8005651185636127604, "upper_bound": "\\u0006# 4HA4"}, {"distinct_range": 8331622.3623947045, "num_eq": 200, "num_range": 50000000, "upper_bound": "Z\\u0007I$uvs"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 4430846634936180427, "row_count": 7243032468914788925}]':::JSONB;
    rsg_test.go:786: CREATE TYPE rand_typ_0 AS (d JSONB, xqc INTERVAL, ca FLOAT8, xekf DECIMAL, uneqq DATE);
    rsg_test.go:786: CREATE TYPE rand_typ_1 AS (wspwl INT8, f GEOMETRY, yb BIT(0), yabx TIMETZ, llcu REFCURSOR, wr FLOAT8);
    rsg_test.go:786: CREATE TYPE rand_typ_2 AS (g INTERVAL, att INT8);
    rsg_test.go:786: SET sql_safe_updates = false;;
    rsg_test.go:786: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:788: 
    rsg_test.go:789: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith4227028768
--- FAIL: TestRandomSyntaxSQLSmith (302.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154601 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-foundations branch-release-24.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/tests/rsg_test.go
+++ b/pkg/sql/tests/rsg_test.go
@@ -61,6 +61,17 @@ func verifyFormat(sql string) error {
 		// Cannot serialize a statement list without parsing it.
 		return nil //nolint:returnerrcheck
 	}
+	for _, stmt := range stmts {
+		if _, ok := stmt.AST.(*tree.DoBlock); ok {
+			// We currently don't handle well identifiers with double quotes in
+			// PLpgSQL blocks, so it's likely that this function will fail for
+			// this statement, thus, we skip the check if we see a DO block
+			// (which can only be a top-level AST node).
+			// TODO(#126727): remove this when double-quoted identifiers are
+			// fixed.
+			return nil
+		}
+	}
 	formattedSQL := stmts.StringWithFlags(tree.FmtShowPasswords)
 	formattedStmts, err := parser.Parse(formattedSQL)
 	if err != nil {
@@ -629,6 +640,7 @@ var ignoredErrorPatterns = []string{
 	"AS OF SYSTEM TIME: timestamp before 1970-01-01T00:00:00Z is invalid",
 	"BACKUP for requested time  needs option 'revision_history'",
 	"RESTORE timestamp: supplied backups do not cover requested time",
+	"a partial index that does not contain all the rows needed to execute this query",
 
 	// Numeric conditions
 	"exponent out of range",

